electrum/transaction.py

```diff
@@ -88,7 +88,19 @@ class MissingTxInputAmount(Exception):
     pass
 
 
-SIGHASH_ALL = 1
+class Sighash(IntEnum):
+    ALL = 1
+    NONE = 2
+    SINGLE = 3
+    ANYONECANPAY = 0x80
+
+    @classmethod
+    def is_valid(cls, sighash) -> bool:
+        for flag in Sighash:
+            for base_flag in [Sighash.ALL, Sighash.NONE, Sighash.SINGLE]:
+                if (flag & ~0x1f | base_flag) == sighash:
```

Comment:
Where is this code from? Took me some time to understand what it is doing...

```python
>>> byt = int.to_bytes(~0x1f, byteorder="big", length=1, signed=True)
>>> int.from_bytes(byt, byteorder="big", signed=False)
224
>>> bin(224)
'0b11100000'
```

Reply:
https://github.com/bitcoin-sv/bitcoin-sv/blob/master/src/script/sighashtype.h
I implemented a similar logic.

Reply:
That link is not really reassuring though as it is for an altcoin's repository...
Does upstream bitcoin/bitcoin have the same logic?